pkg/sql/pgwire/types.go

```diff
@@ -313,7 +314,9 @@ func (b *writeBuffer) writeTextColumnarElement(
 		b.writeLengthPrefixedString(d.String())
 
 	case types.BytesFamily:
-		writeTextBytes(b, string(vecs.BytesCols[colIdx].Get(rowIdx)), conv)
+		writeTextBytes(
+			b, unsafeBytesToString(vecs.BytesCols[colIdx].Get(rowIdx)), conv,
```

Comment:
i don't really know what the contracts are with the `columnar` write thing. do we know that this is safe?

Reply:
I'm almost certain, yes. The deal here is that we're subslicing a bigger slice, which is fine in that that slice is not changing during this call (if it were, we'd have a race). As far as I can tell, `writeTextBytes` does not retain the string after it returns; it appends its bytes to an underlying buffer.